### PR TITLE
Improvement: updating scoreboard and tab list less often

### PR DESCRIPTION
**Pull request by hannibal002**
## What
Should improve performance
Updating scoreboard and tab list only after receiving the update packet, not all the time.

## Changelog Improvements
+ Improved the performance of scoreboard and tab list reading. - hannibal2

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/data/ScoreboardData.kt [hunk 1]
```diff
@@ -66,8 +68,19 @@ class ScoreboardData {
         var objectiveTitle = ""
     }
 
+    var dirty = false
+
+    @SubscribeEvent(receiveCanceled = true)
+    fun onChatPacket(event: PacketEvent.ReceiveEvent) {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/data/ScoreboardData.kt]:
```suggestion
    fun onPacketReceive(event: PacketEvent.ReceiveEvent) {
```

File: src/main/java/at/hannibal2/skyhanni/utils/TabListData.kt [hunk 2]
```diff
@@ -106,9 +108,19 @@ object TabListData {
         return result.dropLast(1)
     }
 
+    var dirty = false
+
+    @SubscribeEvent(receiveCanceled = true)
+    fun onChatPacket(event: PacketEvent.ReceiveEvent) {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/utils/TabListData.kt]:
```suggestion
    fun onPacketReceive(event: PacketEvent.ReceiveEvent) {
```

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
scoreboard seemed to be checked 10x less and tablist 2x less
just these two naming changes

**Comment by hannibal002** [comment]
image
found the reason for my wrong copying

**Review by CalMWolfs — APPROVED** [review]

**Pull request merged**